blockchain/cryptoProvider/signer.go

```diff
@@ -19,27 +19,31 @@ var (
 )
 
 // signer contains the primitives used to correctly sign a transaction
-type signer struct{}
+type signer struct {
+	serializeForSigningHandle func(msg []byte) ([]byte, error)
+}
 
-// NewXSigner will create a new instance of signer
-func NewXSigner() *signer {
-	return &signer{}
+// NewSigner will create a new instance of signer
+func NewSigner() *signer {
+	return &signer{
+		serializeForSigningHandle: serializeForSigning,
```

Comment:
why did we need this handler? It is so hard to test the serializeForSigning function in the test? It does only some converts, byte slice appends and a hash.Compute operation

Reply:
indeed, it was not necessary. removed